recipes/hdf4/build.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+if [[ $(uname) == Darwin ]]; then
+  export DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib
```

Comment:
Sorry. Terrible modify-and-rebase habit: See https://travis-ci.org/conda-forge/staged-recipes/jobs/123373173 and https://github.com/conda-forge/staged-recipes/commit/6d6a43177b3420a405aee22e9891e0d40a0ab4ab#diff-ac97d17248749ce6118a1ed1af9c3820R28